src/ui/steps/IOStep.h

```diff
@@ -0,0 +1,28 @@
+//
+// Created by Anton Ovcharenko on 21.01.2022.
+//
+
+#ifndef TASKMANAGER_SRC_UI_STEPS_IOSTEP_H_
+#define TASKMANAGER_SRC_UI_STEPS_IOSTEP_H_
+
+#include "Step.h"
+#include "io/AbstractReader.h"
+#include "io/AbstractPrinter.h"
+
+class IOStep : public Step {
+public:
+    IOStep(std::shared_ptr<AbstractReader> reader, std::shared_ptr<AbstractPrinter> printer);
```

Comment:
So you moved it out of interface, but you still make all your steps use reader/printer.
Are *all* children use both reader and printer?

Reply:
Almost all, and all use processResult helper function in Step::genNextStep(), which calls printer if the message is not empty. Some steps are special, like QuitStep that does not read or print anything. It could inherit from Step directly. HelpStep only prints, etc.

Reply:
I made an alternative to IOStep — PrinterStep, which carries only printer dependency, and about half of the steps now inherit from it.